functions/cors/index.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2018, Google, LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+// [START functions_cors]
+exports.corsEnabledFunction = (req, res) => {
+  // Set CORS headers for preflight requests
+  // e.g. allow GETs from any origin with the Content-Type header
```

Comment:
Can/should we put these headers in one big `if` block, instead of removing them when they don't apply?

cc @stew-r